app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,352 @@
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+import mozilla.components.browser.storage.sync.Tab as SyncTab
+
+/**
+ * Top-level list UI for displaying Synced Tabs in the Tabs Tray.
+ *
+ * @param syncedTabs The tab UI items to be displayed.
+ * @param onTabClick The lambda for handling clicks on synced tabs.
+ */
+@Composable
+fun SyncedTabsList(syncedTabs: List<SyncedTabsListItem>, onTabClick: (SyncTab) -> Unit) {
+    val listState = rememberLazyListState()
+
+    LazyColumn(
+        modifier = Modifier.fillMaxSize(),
+        state = listState,
+    ) {
+        items(syncedTabs) { syncedTabItem ->
+            when (syncedTabItem) {
+                is SyncedTabsListItem.Device -> SyncedTabsDeviceItem(deviceName = syncedTabItem.displayName)
+                is SyncedTabsListItem.Error -> SyncedTabsErrorItem(
+                    errorText = syncedTabItem.errorText,
+                    errorButton = syncedTabItem.errorButton
+                )
+                is SyncedTabsListItem.NoTabs -> SyncedTabsNoTabsItem()
+                is SyncedTabsListItem.Tab -> {
+                    SyncedTabsTabItem(
+                        tabTitleText = syncedTabItem.displayTitle,
+                        url = syncedTabItem.displayURL,
+                    ) {
+                        onTabClick(syncedTabItem.tab)
+                    }
+                }
+            }
+        }
+
+        item {
+            // The Spacer here is to act as a footer to add padding to the bottom of the list so
+            // the FAB or any potential SnackBar doesn't overlap with the items at the end.
+            Spacer(Modifier.height(240.dp))
+        }
+    }
+}
+
+/**
+ * Text header for sections of synced tabs
+ *
+ * @param deviceName The name of the user's device connected that has synced tabs.
+ */
+@Composable
+fun SyncedTabsDeviceItem(deviceName: String) {
+    Column(Modifier.fillMaxWidth()) {
+        PrimaryText(
+            text = deviceName,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp, top = 16.dp, end = 8.dp, bottom = 8.dp),
+            fontSize = 14.sp,
+            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+            maxLines = 1
+        )
+
+        Divider(color = FirefoxTheme.colors.borderPrimary)
+    }
+}
+
+/**
+ * Synced tab list item UI
+ *
+ * @param tabTitleText The tab's display text.
+ * @param url The tab's URL.
+ * @param onClick The click handler when this synced tab is clicked.
+ */
+@Composable
+fun SyncedTabsTabItem(tabTitleText: String, url: String, onClick: () -> Unit) {
+    Column(
+        modifier = Modifier
+            .clickable(
+                onClickLabel = tabTitleText,
+                onClick = onClick
+            )
+            .padding(horizontal = 16.dp)
+            .defaultMinSize(minHeight = 56.dp),
```

Comment:
General question outside of the review, I am wondering how we decide to uses `defaultMinSize` and  `height(IntrinsicSize.Min)` (in general, nothing in particular about the current code).

Reply:
I used `defaultMinSize` to be a more direct 1:1 to `android:minHeight` in XML (to help ensure with a default text size that there would be enough padding), and the `height(IntrinsicSize.Min)` was used when I was specifically trying out random things to get the dashed border for the error UI to show. The documentation for `Modifier.height()` and `Modifier.defaultMinSize()` aren't the clearest though on when to use either.